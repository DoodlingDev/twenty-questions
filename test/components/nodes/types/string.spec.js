import React from "react";
import FormNodeString from "../../../../src/components/nodes/types/string";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe("rendering", () => {
  it("renders string types correctly", () => {
    const render = renderer.create(
      <FormNodeString
        name="test_string"
        label="test string"
        path="test_form.test_string"
        title="string title"
        description="A simple String"
        valueManager={{
          update: () => {},
          values: {},
          validate: {},
        }}
        />
    ).toJSON();
    expect(render).toMatchSnapshot();
  });
});
