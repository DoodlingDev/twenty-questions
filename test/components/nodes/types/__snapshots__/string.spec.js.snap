// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders string types correctly 1`] = `
<div
  className="q20_node__string"
>
  <span
    className="q20_nodeString__test_string--description"
    htmlFor="test_form.test_string"
  >
    A simple String
  </span>
  <input
    className="q20_nodeString__test_string--input"
    id="test_form.test_string"
    type="text"
  />
</div>
`;
